List and display views

cristianvasquez opened the pull request:
This is mainly a refactor to use views instead of cubes.  

It's mostly taking out references to a cube in the visual components and using views instead.

It also removes the main component, CubeViewer.vue, and adds two new components with distinct responsibilities: 

1. Viewer.vue that accepts a `raw` view,  or cube. And prepares a view (with the corresponding projection and filters, if it applies)   
2. TabularView.vue knows how to render a view as a table.

I also added an rdf-editor to debug the current cube-view.

This depends on the changes to rdf-cube-query at: https://github.com/zazuko/rdf-cube-view-query/pull/75

cristianvasquez committed:
chore: refactor previous to implement

cristianvasquez committed:
chore: refactor get controls from view

cristianvasquez committed:
chore: refactor get controls from view

cristianvasquez committed:
chore: refactor get controls from view

cristianvasquez committed:
chore: update rdf-cube-view-query lib

cristianvasquez committed:
feat: show debug area

cristianvasquez committed:
refactor: no Cube dependency

cristianvasquez committed:
refactor: clownface instead of cube

cristianvasquez committed:
refactor: no cube dependency

cristianvasquez committed:
feat: use RDF-editor

cristianvasquez committed:
feat: add observation constraint

cristianvasquez committed:
refactor: use only views

cristianvasquez committed:
chore: enable es-lint

tpluscode reviewed (changes requested):
A lot going on here.

On top of some technical comments, I have one observation about strictly separating cubes and views. I thought that they would be quite similar in how they are consumed. After all, both are just a different way for having the view query library fetch the tabular data.

In other words, I would expect that once selected, the rest of the code could be pretty uniform regardless of whether cube of view is being handled

cristianvasquez commented:
> A lot going on here.
> 
> On top of some technical comments, I have one observation about strictly separating cubes and views. I thought that they would be quite similar in how they are consumed. After all, both are just a different way for having the view query library fetch the tabular data.
> 
> In other words, I would expect that once selected, the rethe st of the code could be pretty uniform regardless of whether cube of view is being handled
The Vue components of this application are coupled with the domain objects of the query library, and before the refactor, most of the components expected `Cube objects` as input. I imagine it's because the app was built before the concept of views.

The idea is now to show View, which takes data from many cubes. The refactor is, in fact, an effort to use the superset, the View, precisely to have (a little more) uniformity.

I would use a plain model of rows and cells that only shows tabular data ;)

cristianvasquez commented on src/components/ObservationValue.vue:
```diff
@@ -1,13 +1,13 @@
     <resource-explorer
       :uri="value"
-      :cube="cube"
+      :clownface="clownface"
```
that's right :)

cristianvasquez commented on src/components/DimensionHeader.vue:
```diff
@@ -107,15 +102,15 @@ export default defineComponent({
   computed: {
     label () {
-      return this.dimension.out(ns.schema.name, { language: this.language }).value
+      return this.dimension.ptr.out(ns.schema.name, { language: this.language }).value
```
yes, `out` in the dimension object proxies ptr.out. I've changed this to try to depend on the cf pointer model instead of the cf pointer model + dimension model

cristianvasquez commented on src/components/rdf/Editbox.vue:
```diff
@@ -0,0 +1,80 @@
+import { defineProps, ref } from 'vue'
+
+const props = defineProps({
+  format: {
+    required: true,
+    default: 'text/turtle',
+    type: String
+  },
```
I ignore how to use it as I'm adding this as a draft input interface for triples. 

I ignore how to get the quads of the rdf-editor Web-component from the Vue page; how do you do that?

cristianvasquez committed:
feat: debug-box

cristianvasquez committed:
chore: refactor clownface -> pointer

cristianvasquez committed:
refactor: rename resource -> pointer

cristianvasquez committed:
feat: added a 'Debug box'

cristianvasquez committed:
feat: added a 'Debug box'

cristianvasquez committed:
feat: more information in debug box
feat: accept views with more than one cube
feat: apply defaults for cubes without page limits

cristianvasquez committed:
feat: more information in debug box
feat: accept views with more than one cube
feat: apply defaults for cubes without page limits
